### PR TITLE
[Feature] Move company login to landing page

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

- Removed `For companies` button from login page
- Trigger company login flow from `Enterprise log in` in landing page

### CODE REVIEW

**Comment by David-Henner** [comment]
Should we still take into account `AuthenticationFeatureProvider.allowDirectCompanyLogin` flag in the updated landing screen ?

File: Wire-iOS/Sources/Authentication/Landing/LandingViewController.swift [hunk 1]
```diff
@@ -431,7 +432,8 @@ class LandingViewController: AuthenticationStepViewController {
     }
     
     @objc public func enterpriseLoginButtonTapped(_ sender: AnyObject!) {
-        // TODO: Show enterprise login popup
+        // Do we need to log this into Analytics ??
```
Comment (typfel) [inline Wire-iOS/Sources/Authentication/Landing/LandingViewController.swift]: We don't have any analytics atm.

File: Wire-iOS/Sources/Authentication/AuthenticationInterfaceBuilder.swift [hunk 2]
```diff
@@ -90,14 +90,7 @@ class AuthenticationInterfaceBuilder {
             return viewController
 
         case .provideCredentials(let credentialsFlowType, let prefill):
-            let viewController = makeCredentialsViewController(for: .login(credentialsFlowType, prefill))
-
-            // Add the item to start company login if needed.
-            if featureProvider.allowDirectCompanyLogin {
```
Comment (typfel) [inline Wire-iOS/Sources/Authentication/AuthenticationInterfaceBuilder.swift]: We don't need to check `featureProvider.allowDirectCompanyLogin` at the new place? Can we remove this property then?

**Comment by typfel** [comment]
> Should we still take into account `AuthenticationFeatureProvider.allowDirectCompanyLogin` flag in the updated landing screen ?

Yes, think we should still it into account and hide the button if that feature is disabled.

**Pull request merged**